src/factor_graph.h

```diff
@@ -175,8 +173,17 @@ class CompactFactorGraph {
 };
 
 inline double CompactFactorGraph::potential(
-    const CompactFactor& factor, const variable_value_t assignments[]) {
-  return factor.potential(vifs.get(), assignments, -1, -1);
+    const CompactFactor& factor, const variable_value_t assignments[],
+    factor_value_t factor_value) {
+  // For boolean, this is stored in CompactFactor.value;
+  //     caller shouldn't set the factor_value arg.
+  // For categorical, this is stored in Factor.factor_params.value.
+  //     caller should pass in via the factor_value arg.
+  // TODO: better data structures than the factor_params duct tape...
+  if (factor_value == Factor::DEFAULT_VALUE) {
+    factor_value = factor.value;
+  }
```

Comment:
I'm ok with that =) (Sorry for being terse, was on the treadmill :) )

On Sun, Jul 17, 2016 at 10:05 AM alldefector <notifications@github.com>
wrote:

> In src/factor_graph.h
> <https://github.com/HazyResearch/sampler/pull/42#discussion_r71085668>:
>
> > @@ -175,8 +173,17 @@ class CompactFactorGraph {
> >  };
> >
> >  inline double CompactFactorGraph::potential(
> > -    const CompactFactor& factor, const variable_value_t assignments[]) {
> > -  return factor.potential(vifs.get(), assignments, -1, -1);
> > +    const CompactFactor& factor, const variable_value_t assignments[],
> > +    factor_value_t factor_value) {
> > +  // For boolean, this is stored in CompactFactor.value;
> > +  //     caller shouldn't set the factor_value arg.
> > +  // For categorical, this is stored in Factor.factor_params.value.
> > +  //     caller should pass in via the factor_value arg.
> > +  // TODO: better data structures than the factor_params duct tape...
> > +  if (factor_value == Factor::DEFAULT_VALUE) {
> > +    factor_value = factor.value;
> > +  }
>
> How about factor.feature_value?
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/HazyResearch/sampler/pull/42/files/73792183d31eac97322ce3c5dd89223230078f47#r71085668>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AHPtuDklu7-ZguYS-19cv9fyWKsbtAh1ks5qWmDegaJpZM4JN8j9>
> .
>